From 28df5fea345811c8372856bbc9b79523cef5a69e Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Thu, 5 Oct 2017 16:35:32 +0200
Subject: [PATCH 4/7] ccache: use different bind mount directory

When using nspawn /tmp gets over-mounted with tmpfs. Let's use different
directory for our ccache bind mount.

(cherry picked from commit d75783cf16d9ac657119b6b8521bb3491018c921)
---
 etc/mock/site-defaults.cfg     | 2 +-
 py/mockbuild/plugins/ccache.py | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/etc/mock/site-defaults.cfg b/etc/mock/site-defaults.cfg
index 3ee870d..c24f623 100644
--- a/etc/mock/site-defaults.cfg
+++ b/etc/mock/site-defaults.cfg
@@ -125,7 +125,7 @@
 # config_opts['plugin_conf']['root_cache_opts']['compress_program'] = "pigz"
 # config_opts['plugin_conf']['root_cache_opts']['extension'] = ".gz"
 # config_opts['plugin_conf']['root_cache_opts']['exclude_dirs'] = ["./proc", "./sys", "./dev",
-#                                                                  "./tmp/ccache", "./var/cache/yum" ]
+#                                                                  "./var/tmp/ccache", "./var/cache/yum" ]
 #
 # bind mount plugin is enabled by default but has no configured directories to
 # mount
diff --git a/py/mockbuild/plugins/ccache.py b/py/mockbuild/plugins/ccache.py
index 23ec21f..3d58620 100644
--- a/py/mockbuild/plugins/ccache.py
+++ b/py/mockbuild/plugins/ccache.py
@@ -33,7 +33,7 @@ class CCache(object):
         buildroot.preexisting_deps.append("ccache")
         plugins.add_hook("prebuild", self._ccacheBuildHook)
         plugins.add_hook("preinit", self._ccachePreInitHook)
-        buildroot.mounts.add(BindMountPoint(srcpath=self.ccachePath, bindpath=buildroot.make_chroot_path("/tmp/ccache")))
+        buildroot.mounts.add(BindMountPoint(srcpath=self.ccachePath, bindpath=buildroot.make_chroot_path("/var/tmp/ccache")))
 
     # =============
     # 'Private' API
@@ -50,11 +50,11 @@ class CCache(object):
     @traceLog()
     def _ccachePreInitHook(self):
         getLog().info("enabled ccache")
-        envupd = {"CCACHE_DIR": "/tmp/ccache", "CCACHE_UMASK": "002"}
+        envupd = {"CCACHE_DIR": "/var/tmp/ccache", "CCACHE_UMASK": "002"}
         if self.ccache_opts.get('compress') is not None:
             envupd["CCACHE_COMPRESS"] = str(self.ccache_opts['compress'])
         self.buildroot.env.update(envupd)
 
-        mockbuild.util.mkdirIfAbsent(self.buildroot.make_chroot_path('/tmp/ccache'))
+        mockbuild.util.mkdirIfAbsent(self.buildroot.make_chroot_path('/var/tmp/ccache'))
         mockbuild.util.mkdirIfAbsent(self.ccachePath)
         self.buildroot.uid_manager.changeOwner(self.ccachePath, recursive=True)
-- 
2.18.0

